src/main/java/io/cryostat/messaging/WsClient.java

```diff
@@ -83,13 +88,46 @@ String readMessage() {
     void writeMessage(String message) {
         if (!this.sws.isClosed()) {
             try {
+                WsMessageEmitted evt =
+                        new WsMessageEmitted(
+                                sws.remoteAddress().host(),
+                                sws.remoteAddress().port(),
+                                sws.uri(),
+                                message.length());
+                evt.begin();
+
                 this.sws.writeTextMessage(message);
```

Comment:
If `writeTextMessage` here throws an Exception, the event object will be instantiated and `begin()`'d, but never committed. I think we would want to capture some info about the exception - even just a boolean flag indicating whether an exception occurred here or not might be sufficient.

The pattern to employ here looks roughly like this:

```
WsMessageEmitted evt = new WsMessageEmitted(...);
evt.begin();
try {
  sws.writeTextMesage(message);
} catch (Exception e) {
  logger.warn(e);
  evt.setExceptionStatus(true); // assuming we default the state to false when the event is created
} finally {
  evt.end();
  if (evt.shouldCommit()) {
    evt.commit();
  }
}
```

This way the event is guaranteed to always be initialized, and always check if it should commit and then commit if so, whether an exception occurs or not. This is a good pattern to follow in general if there is any chance at all that an exception or error can occur between event instantiation and event commit - and there effectively always is, since something like an OutOfMemoryError could happen at any point and at any time.